File: AngularBooking/Models/Showing.cs
using Microsoft.AspNet.OData.Query;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AngularBooking.Models
{
    public class Showing : IModel
    {
        public int Id { get; set; }
        public int EventId { get; set; }
        public int RoomId { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public int PricingStrategyId { get; set; }

        // navigation
        public Event Event { get; set; }
        public Room Room { get; set; }
        public PricingStrategy PricingStrategy { get; set; }

        [NotExpandable]
        public List<Booking> Bookings { get; set; }

    }
}

File: AngularBooking/ClientApp/src/app/core/_models/entity/showing.ts
import { Event } from "./event";
import { Room } from "./room";
import { PricingStrategy } from "./pricing-strategy";

export class Showing {

  constructor(
    public id?: number,
    public eventId?: number,
    public roomId?: number,
    public startTime?: string,
    public endTime?: string,
    public pricingStrategyId?: number,
    public event?: Event,
    public room?: Room,
    public pricingStrategy?: PricingStrategy
  ) { }
}
